File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentMemoryRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentMemoryModel;

public interface IAgentMemoryRepository {
    CompletableFuture<AgentMemoryModel> getAsync(UUID agentId, String key);
    CompletableFuture<List<AgentMemoryModel>> listAsync(UUID agentId);
    CompletableFuture<Void> upsertAsync(UUID agentId, String key, String content);
    CompletableFuture<Boolean> deleteAsync(UUID agentId, String key);
}

File: apps/backend/src/Features/Context/Domain/IAgentMemoryRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Context;

public interface IAgentMemoryRepository
{
    Task<AgentMemoryRecord?> GetByAsync(AgentMemoryFilter filter, CancellationToken ct = default);
    Task<IReadOnlyList<AgentMemoryRecord>> ListAsync(Guid agentId, CancellationToken ct = default);
    Task UpsertAsync(Guid agentId, string key, string content, CancellationToken ct = default);
    Task<bool> DeleteAsync(Guid agentId, string key, CancellationToken ct = default);
}
